File: python/hackerrank/migratory_birds.py
# https://www.hackerrank.com/challenges/migratory-birds/problem

def migratoryBirds(arr):
    types = {1:0, 2:0, 3:0, 4:0, 5:0}
    for el in arr:
        types[el] += 1

    result = 1
    for type_, count in types.items():
        if types[result] == count and type_ > result:
            continue
        
        if types[result] < count:
            result = type_
            
    return result

File: go/hackerrank/migratoryBirds.go
# https://www.hackerrank.com/challenges/migratory-birds/problem

func migratoryBirds(arr []int32) int32 {
    types := map[int32]int32{1:0, 2:0, 3:0, 4:0, 5:0}
    for _, el := range arr {
        types[el] += 1
    }

    var result int32 = 1
    for t, v := range types {
        fmt.Println(t,v, types[result])
        if types[result] == v && t > result {
            continue
        }

        if (types[result] < v) {
            result = t
        }
    }

    return result
}